app/assets/javascripts/govuk-component/option-select.js

```diff
@@ -0,0 +1,137 @@
+(function ($) {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  function OptionSelect(options){
+    /* This JavaScript provides two fucntional enhancements to option-select components:
+      1) A link that clears all of the checkboxes (referred to here as the "clearing link")
+      2) Open/closing of the list of checkboxes - this is not provided for ie6 and 7 as the performance is too janky.
+    */
+
+    this.$optionSelect = options.el;
+    this.$options = this.$optionSelect.find("input[type='checkbox']");
+
+    // Build clearing link
+    this.$clearingLink = this.attachClearingLink();
+    this.updateClearingLink();
+
+    // Attach event listeners for clearing
+    this.$clearingLink.on('click', this.resetOptions.bind(this));
+    this.$options.on('click', this.updateClearingLink.bind(this));
+
+    // Performance in ie 6/7 is not good enough to support animating the opening/closing
+    // so do not allow option-selects to be collapsible in this case
+    var allowCollapsible = (typeof ieVersion == "undefined" || ieVersion > 7) ? true : false;
+    if(allowCollapsible){
+
+      // Add js-collapsible class to parent for CSS
+      this.$optionSelect.addClass('js-collapsible');
+
+      // Add the little arrow toggle image
+      this.attachOpenCloseToggle();
+
+      // Add open/close listeners
+      this.$optionSelect.find('.container-head').on('click', this.toggleOptionSelect.bind(this));
```

Comment:
The test I'd apply is, if a developer thinks about changing or removing that class for styling purposes, can they know for certain they won't break any Javascript. So any hook for Javascript should probably be separate and prefixed with js-